apps/api/pages/api/event-types/by-slug/[slug]/_auth-middleware.ts

```diff
@@ -0,0 +1,33 @@
+import type { NextApiRequest } from "next";
+
+import { HttpError } from "@calcom/lib/http-error";
+
+import { schemaQuerySlug } from "~/lib/validations/shared/querySlug";
+
+async function authMiddleware(req: NextApiRequest) {
+  const { userId, isAdmin, prisma } = req;
+  const { slug, ownerId } = schemaQuerySlug.parse(req.query);
+
+  const effectiveUserId = isAdmin && ownerId ? ownerId : userId;
+
+  const eventType = await prisma.eventType.findFirst({
+    where: { slug },
+    include: {
+      users: {
+        select: { id: true },
+      },
+    },
+  });
+
+  if (!eventType) {
+    throw new HttpError({ statusCode: 404, message: "Event type not found" });
+  }
+
+  const isAuthorized = isAdmin || eventType.users.some((user) => user.id === effectiveUserId);
+
+  if (!isAuthorized) {
+    throw new HttpError({ statusCode: 403, message: "Forbidden" });
+  }
+}
+
+export default authMiddleware;
```

Comment:
I refactored this file in commit [feat: allow admins to fetch event types on behalf of others](https://github.com/calcom/cal.com/pull/13374/commits/6765cc7b049607234b4440bbe436a1aa0931ac0d):
1. Before you could pass non existing slug name and the "authMiddleware" would pass without errors. Now it throws 404.
2. If user is not admin and his / her user Id is not part of event type, then throw 403 Forbidden. Also, using effectiveUserId allow admins to fetch event types on behalf of others. Done in commit [feat: allow admins to fetch event types on behalf of others](https://github.com/calcom/cal.com/pull/13374/commits/6765cc7b049607234b4440bbe436a1aa0931ac0d)